superset/common/query_object_factory.py

```diff
@@ -99,6 +112,30 @@ def _process_row_limit(
         )
         return apply_max_row_limit(row_limit or default_row_limit)
 
+    def _process_time_range(
```

Comment:
This look good. It would be nice to define this as a staticmethod or separate function, so it's easier to test, and also add some unit tests. It's hard to understand what's going on here and what are the differnet contexts where this is running.

Reply:
Thanks for your review. I have added code comments and also unit tests.